### ISSUE TITLE
Wii.sc class triggering multiple unwanted actions on button press

### ISSUE THREAD

**Issue by rev-** [supercollider/supercollider#1196]
Working with a WiiMote I found an issue: pressing any button would trigger _all_ the actions defined for any of the buttons on the WiiMote.
Found a workaround working on my machine, seeming to be universal, may need optimization.

Code affected: file:  supercollider / SCClassLibrary / Common / Control / WII.sc ; starting line 426 
//--BEGIN 

prHandleButtonEvent{ |buttonData|
      // buttonData are bits that decode to separate buttons (do in Primitive internally, and pass on Array?
      //("handle button Event"+buttonData).postln;
      remote_buttons = buttonData;
      [ \bA, \bB, \bOne, \bTwo, \bMinus, \bHome, \bPlus, \bUp, \bDown, \bLeft, \bRight ].do{ |key|
                   actionSpec.at( key ).value( spec.at(key).value );
                   if ( dumpEvents, { (key + spec.at(key).value.round(0.00001)).postln; });
      }
}

//--END

Workaround code solving the issue (sostituting the former)

//--BEGIN

prHandleButtonEvent{ |buttonData|
                //Receive the buttonData array and parse it in order to find which buttons changed their states. Then, update their states and send requested actions.

```
            [ \bA, \bB, \bOne, \bTwo, \bMinus, \bHome, \bPlus, \bUp, \bDown, \bLeft, \bRight ].do{ |key,i|
                    if ( buttonData[i]!=remote_buttons[i], { remote_buttons[i]=buttonData[i]; actionSpec.at( key ).value( spec.at(key).value ); });
                    if ( dumpEvents, { (key + spec.at(key).value.round(0.00001)).postln; });
            }
    }
```

//--END

Basically before throwing actions, it parses the buttonData array and find the buttons for which state was changed, then goes on putting new values in remote_buttons array and throwing only the actions needed.
My programming skill is poor, so it's likely someone would need to provide some changes. What caused the problem in the original version was the acceptance of the whole buttonData array and the subsequent launch of the actions defined for any button, not only those whose state actually changed.

Regards,
rev

**Comment by brianlheim** [comment]
Am I correct in thinking that we're no longer supporting Wii.sc features (deprecated in #2698) and can close this?

**Comment by vivid-synth** [comment]
Yeah, let's close. We can revive if someone wants to make a Wiimote quark
